docs/parallel-mode.md

```diff
@@ -0,0 +1,76 @@
+# Parallel Pull and Unpack Mode
+
+## Introduction
+
+The SOCI Snapshotter's lazy loading technology has proven effective in accelerating container startup times. The technology defers the full image download and unpacking until the container actually needs the files, allowing for near-instant container launches.
+
+However, some users prefer the traditional upfront loading model, where the entire container image is pulled and prepared before the container is started. This is particularly the case for I/O-bound applications that require fast access to the full rootfs.
+
+To support users who favor the upfront loading workflow, we have introduced a new _parallel-pull-unpack_ mode in the SOCI Snapshotter. This mode enables concurrent processing of container image layers, significantly improving the performance of the upfront loading process.
+
+## Functionality
+ 
+The _parallel-pull-unpack_ mode operates as follows:
+
+* Parallel Layer Downloads: When preparing a container image snapshot, the snapshotter initiates concurrent HTTP range requests to fetch the individual image layers. Each layer is downloaded in parallel using multiple HTTP connections.
```

Comment:
Specify range GET? And maybe mention buffering to a temp file, so that users know that we are not buffering it in memory but instead to a file on disk.